### PR TITLE
Allow fetching additional k8s resource to prometheus health subsystem

### PR DESCRIPTION
**Pull request by rawagner**
Ceph plugin needs to fetch k8s resource to determine the name of ceph cluster - https://github.com/openshift/console/pull/2411

@anmolsachan please let me know if it works for you. K8s resource is passed to `healthHandler` function. I've also updated `demo-plugin` to showcase the usage

### CODE REVIEW

**Comment by rawagner** [comment]
@vojtechszocs please take a look

**Comment by vojtechszocs** [comment]
> I've also updated `demo-plugin` to showcase the usage

:+1:

File: frontend/packages/console-plugin-sdk/src/typings/dashboards.ts [hunk 1]
```diff
@@ -37,12 +34,24 @@ namespace ExtensionProperties {
...
+    extends DashboardsOverviewHealthSubsystem {
     /** The Prometheus query */
     query: string;
+
+    /** Resource which will be fetched and passed to healthHandler  */
+    resource?: FirehoseResource;
+
+    /** Resolve the subsystem's health */
+    healthHandler: (
+      response,
```
Comment (vojtechszocs) [inline frontend/packages/console-plugin-sdk/src/typings/dashboards.ts]: Are we OK with `response` having `any` type here?

(`DashboardsOverviewHealthURLSubsystem` allows response type parametrization via generics.)

File: frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx [hunk 2]
```diff
@@ -88,42 +93,64 @@ const HealthCard_ = connect(mapStateToProps)(({
...
+  [
+    watchURL,
+    stopWatchURL,
+    watchPrometheus,
+    stopWatchPrometheusQuery,
+    watchAlerts,
+    stopWatchAlerts,
+    watchK8sResource,
+    stopWatchK8sResource,
+    stopWatchAlerts, JSON.stringify(flags),
```
Comment (vojtechszocs) [inline frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx]: Doesn't ESLint complain about two array elements on the same line?

Comment (rawagner) [reply frontend/public/components/dashboards-page/overview-dashboard/health-card.tsx]: it does not, core does not have such strong checking as plugins.
Moved to new line

File: frontend/packages/console-plugin-sdk/src/typings/dashboards.ts (hunk 1 above)
Comment (rawagner) [reply frontend/packages/console-plugin-sdk/src/typings/dashboards.ts]: it is actually a `PrometheusResponse` type, added

**Comment by anmolsachan** [comment]
@rawagner I am observing the following issue on overview dashboard:

The overview dashboard also shows the subsystem error instead of showing the OCP cluster status, When the subsystem is not available : 
Screenshot from 2019-08-28 18-54-41
Screenshot from 2019-08-28 18-54-47

When the subsystem is available it shows the correct information:
Screenshot from 2019-08-28 19-01-54

**Comment by rawagner** [comment]
That is on purpose. The main health status in an aggregation of all subsystem healths and we always show the worst one (or `Multiple errors` if more subsystems are in error state)

**Pull request merged**